### ISSUE TITLE
arch/arm/crypto/aes-ce-core.S build failure after c61b1607ed4fbbf2ba7c86f29768cff44a1a88f8

### ISSUE THREAD

**Issue by nathanchance** [ClangBuiltLinux/linux#659]
This was reported with Linaro's CI.

```
00:00:56 arch/arm/crypto/aes-ce-core.S: Assembler messages:
00:00:56 arch/arm/crypto/aes-ce-core.S:299: Error: selected processor does not support `movw ip,:lower16:.Lcts_permute_table' in ARM mode
00:00:56 arch/arm/crypto/aes-ce-core.S:300: Error: selected processor does not support `movt ip,:upper16:.Lcts_permute_table' in ARM mode
00:00:56 arch/arm/crypto/aes-ce-core.S:337: Error: selected processor does not support `movw ip,:lower16:.Lcts_permute_table' in ARM mode
00:00:56 arch/arm/crypto/aes-ce-core.S:338: Error: selected processor does not support `movt ip,:upper16:.Lcts_permute_table' in ARM mode
00:00:56 arch/arm/crypto/aes-ce-core.S:552: Error: selected processor does not support `movw ip,:lower16:.Lcts_permute_table' in ARM mode
00:00:56 arch/arm/crypto/aes-ce-core.S:553: Error: selected processor does not support `movt ip,:upper16:.Lcts_permute_table' in ARM mode
00:00:56 arch/arm/crypto/aes-ce-core.S:638: Error: selected processor does not support `movw ip,:lower16:.Lcts_permute_table' in ARM mode
00:00:56 arch/arm/crypto/aes-ce-core.S:639: Error: selected processor does not support `movt ip,:upper16:.Lcts_permute_table' in ARM mode
00:00:56 clang-9: error: assembler command failed with exit code 1 (use -v to see invocation)
```

```
$ make -j$(nproc) -s ARCH=arm CC=clang CROSS_COMPILE=arm-linux-gnueabi- O=out distclean allyesconfig arch/arm/crypto/
```

is enough to reproduce. I bisected it to commit [c61b1607ed4f](https://git.kernel.org/next/linux-next/c/c61b1607ed4fbbf2ba7c86f29768cff44a1a88f8) ("crypto: arm/aes-ce - implement ciphertext stealing for XTS") in -next.

**Comment by nickdesaulniers** [comment]
Does `-march=armv7-a` or newer get set for this config?

**Comment by nathanchance** [comment]
With `allyesconfig`, `-march` is set to `armv6k`. I suppose `-march=armv7-a` should be explicitly set for these files now?

**Comment by nickdesaulniers** [comment]
potentially.

@ardbiesheuvel @smithp35 do you know whether:
1. that addressing mode is supported in armv6 and Clang is wrong or
2. that addressing mode is unsupported in armv7 and we should build with `-march=armv7-a`?

**Comment by ardbiesheuvel** [comment]
Addressed by this patch, I think:
https://lore.kernel.org/linux-crypto/20190917085001.792-1-ard.biesheuvel@arm.com/

**Comment by nathanchance** [comment]
Yes, that patch resolves this issue, thanks :)

**Comment by nickdesaulniers** [comment]
Thanks @ardbiesheuvel !  For me to follow up on 

```
Note that using armv7-a
here may trigger an issue with the upcoming Clang 10 release,
which no longer permits .arch/.fpu combinations it views as
incompatible.
```

Are there cases in kernel sources where Clang is incorrect here?  If so, I'll try to get bugs filed.

**Comment by ardbiesheuvel** [comment]
> Thanks @ardbiesheuvel ! For me to follow up on
> 
> ```
> Note that using armv7-a
> here may trigger an issue with the upcoming Clang 10 release,
> which no longer permits .arch/.fpu combinations it views as
> incompatible.
> ```
> 
> Are there cases in kernel sources where Clang is incorrect here? If so, I'll try to get bugs filed.

The Linaro toolchain team run a CI job using Clang HEAD to build linux -next, and they are the ones that reported this to me. We can do a sweep for crypto-neon-fp-armv8 and ensure that the .arch is set to armv8-a, but other than that, I don't think there's a lot of work to be done here.

**Comment by nickdesaulniers** [comment]
fixed: 3aa6d4abd4ebf07c36e0a9a507af0311fcb267dc 5.4-rc3 (https://git.kernel.org/pub/scm/linux/kernel/git/torvalds/linux.git/commit/?id=3aa6d4abd4ebf07c36e0a9a507af0311fcb267dc)
